ModuleNotFoundError: No module named 'filelock'

Hi maintainers — thanks for the project!

On a clean install, using MOABB fails because the `filelock` dependency is missing from the installation requirements.

### Steps to reproduce
```bash
python3 -m venv moabb-test
source moabb-test/bin/activate
pip install moabb
python -c "import moabb"
```

### Actual behavior
It raises:
>ModuleNotFoundError: No module named 'filelock'

### Expected behavior
`filelock` should be installed automatically as a dependency

### Environment
- OS: Windows 11 + WSL2 (Ubuntu 24.04)
- Python: 3.12.3
- moabb version: 1.4.3

Suggestion: add `filelock` to the package requirements (e.g., `install_requires` / `requirements.txt`, depending on how dependencies are managed).
If you agree with the proposed fix, I’m happy to open a PR adding `filelock` to the project’s declared dependencies. It would be my pleasure to submit my first PR to this project.

Thanks! I've updated pyproject.toml to include `filelock`, based on version currently used by`mne`: #959

Regarding the test case: I believe adding a specific test for this might be redundant because utils.py imports filelock at the top level. This means that without the package installed, the entire test suite (and the library itself) fails to load immediately with an ImportError, which already captures the issue.

For example, running existing tests like pytest moabb/tests/test_verbose.py on a clean environment results in:
```
ImportError while loading conftest '/mnt/d/PROJEKTY/ZPB2/moabb/moabb/tests/conftest.py'.
moabb/__init__.py:4: in <module>
    from .benchmark import benchmark
...
moabb/utils.py:14: in <module>
    import filelock
E   ModuleNotFoundError: No module named 'filelock'
```

The CI/CD pipeline is usually a great way to catch these issues, as the clean environment exposes missing dependencies

--- a/docs/source/whats_new.rst
+++ b/docs/source/whats_new.rst
@@ -46,6 +46,7 @@ API changes
 Requirements
 ~~~~~~~~~~~~
 - Allows CodeCarbon environment variables or a configuration file to be defined in the home directory or the current working directory (:gh:`866` by `Ethan Davis`_).
+- Added ``filelock`` as a core dependency to fix missing import errors in utils (:gh:`959` by `Mateusz Naklicki`_).
 
 Bugs
 ~~~~
@@ -711,3 +712,4 @@ API changes
 .. _Romani Michele: https://github.com/BRomans
 .. _Lionel Kusch: https://github.com/lionelkusch
 .. _Victor Martinez-Cagigal: https://github.com/vicmarcag
+.. _Mateusz Naklicki: https://github.com/luluu9

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -49,6 +49,7 @@ dependencies = [
     "memory-profiler>=0.61.0",
     "edflib-python>=1.0.6",
     "edfio>=0.4.2",
+    "filelock>=3.18.0",
     "pytest>=8.3.5",
     "mne-bids>=0.16",
     "scikit-learn>=1.6",

